server/events/project_command_builder.go

```diff
@@ -673,3 +700,48 @@ func (p *DefaultProjectCommandBuilder) validateWorkspaceAllowed(repoCfg *valid.R
 
 	return repoCfg.ValidateWorkspaceAllowed(repoRelDir, workspace)
 }
+
+func filterProjects(projects []models.Project, filter string) ([]models.Project, error) {
+	trimmedFilter := strings.TrimPrefix(filter, "./")
+	filteredProjects := make([]models.Project, 0, len(projects))
+	for _, proj := range projects {
+		match, err := filepath.Match(trimmedFilter, proj.Path)
+		if err != nil {
+			return nil, err
+		}
+		if match {
+			filteredProjects = append(filteredProjects, proj)
+		}
+	}
+	return filteredProjects, nil
+}
+
+func filterValidProjects(projects []valid.Project, filter string) ([]valid.Project, error) {
```

Comment:
Can we combine both `filterProjects` and `filterValidProjects` ?

Reply:
Only difference I see is `models.Project` and `proj.Path` vs `valid.Project` and `proj.Dir`